addons/binding/org.openhab.binding.toon/src/main/java/org/openhab/binding/toon/handler/ToonBridgeHandler.java

```diff
@@ -122,11 +123,11 @@ private void disposeApiClient() {
 
     private void updateStatus() {
         try {
-            if (configuration == null || configuration.username == null || configuration.username.length() == 0) {
+            if (configuration == null || StringUtils.isEmpty(configuration.username)) {
                 updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Username not configured");
                 return;
             }
-            if (configuration == null || configuration.password == null || configuration.password.length() == 0) {
+            if (StringUtils.isEmpty(configuration.password)) {
```

Comment:
How do you feel about making `configuration == null` it's own case? Would be easier to read and "Username not configured" is misleading if there's no configuration at all (not sure where it comes from.